app/user/rest.py

```diff
@@ -616,6 +623,11 @@ def update_password(user_id):
         raise InvalidRequest(errors, status_code=400)
 
     update_user_password(user, pwd)
+
+    # save login event
+    if login_data is not None and login_data != {}:
```

Comment:
What about `if login_data` style? or you prefer the more explicit version?

Reply:
Nah, I'm just a python newb :D - I'll fix it up!